src/views/activity/activity_item_view.js

```diff
@@ -20,5 +26,26 @@ export default BaseView.extend({
 
   sendSelectionDetails (event) {
     webChannel.sendUrlSelected(this.model.url, 'url');
+  },
+
+  injectFavicon () {
+    // Manually construct the favicon element and inject it favicon into the
+    // DOM, allowing us to delegate events to its onLoad event, which is not
+    // possible through Ampersand since load events do not bubble.
+    const favicon = document.createElement('img');
+    favicon.src = this.model.displayImage();
+    favicon.onload = this.calculateIconColor;
+    this.el.querySelectorAll('.favicon')[0].appendChild(favicon);
+  },
+
+  calculateIconColor (event) {
+    // If able to do so in 20ms or less, calculate the dominant color of the
+    // favicon and set that to the background color of its parent.
+    const START = new Date().getTime();
+    const favicon = new Favicon(event.target);
+    const END = new Date().getTime();
```

Comment:
In testing, this was coming back in <= 1ms for 16x16 icons, and < 10ms for 64x64 :+1: